src/db.c

```diff
@@ -1316,16 +1315,16 @@ int dbSwapDatabases(int id1, int id2) {
 /* Logically, this discards (flushes) the old main database, and apply the newly loaded
  * database (temp) as the main (active) database, the actual freeing of old database
  * (which will now be placed in the temp one) is done later. */
-void swapMainDbWithTempDb(tempDb *tempDb) {
+void swapMainDbWithTempDb(redisDb *tempDb) {
     if (server.cluster_enabled) {
         /* Copy slots_to_keys from tempdb just loaded to main slots_to_keys. */
-        memcpy(server.cluster->slots_to_keys, server.cluster->slots_to_keys_tempdb,
-           sizeof(server.cluster->slots_to_keys_tempdb));
+        memcpy(*server.db->slots_to_keys, *tempDb->slots_to_keys,
+           sizeof(*tempDb->slots_to_keys));
```

Comment:
This code doesn't swap them. It just copies one to the other. Don't we want to swap them like we do for dict, expires, avg_ttl and expires_cursor below?

Also, we don't need memcpy here. We can just swap the pointers, like this:

If we want to swap them, don't we want to do something like this?

```Suggestion
        clusterSlotsToKeysData *tmp = server.db->slots_to_keys;
        server.db->slots_to_keys = tempDb->slots_to_keys;
        tempDb->slots_to_keys = tmp;
```

... or simpler, we can do this for all db numbers in the for loop below. It will be NULL for all DBs except 0, but it does matter IMO.

Reply:
Thank you for the suggestions @zuiderkwast. Commited changes for swapping pointers instead of memcpy and splitting slotToKeyFlush into 2 functions